### ISSUE TITLE
"Unhandled Exception" with no further information

### ISSUE THREAD

**Issue by carcinocron** [martenframework/marten#50]
When running `marten serve --error-trace` on a fresh project, I get the following error:

![Screenshot at 2022-11-20 17-38-21](https://user-images.githubusercontent.com/4094542/202933379-7bc6d13d-e1be-4ddf-97df-438f77c7a1fa.png)

The error is double-printed to the terminal output:

![Screenshot at 2022-11-20 17-38-46](https://user-images.githubusercontent.com/4094542/202933383-1d321c69-508e-45ca-99b5-760eec12b100.png)

I expected `--error-trace`  to show more information. I could not identify a place to put something like `pp error` that worked.

I looked for something like a log file but did not find anything.

I also noticed that `marten serve --error-trace`  does not "hot reload" or "watch" changed files.

Adding my [first handler](https://martenframework.com/docs/getting-started/tutorial#writing-a-first-handler) did not change anything.

more info about my system:
```sh
[forge@nuc7i7bnh sg-uc]$ uname -a
Linux nuc7i7bnh 6.0.7-zen1-1-zen #1 ZEN SMP PREEMPT_DYNAMIC Thu, 03 Nov 2022 18:02:00 +0000 x86_64 GNU/Linux
[forge@nuc7i7bnh sg-uc]$ crystal -v
Crystal 1.6.2 (2022-11-04)

LLVM: 14.0.6
Default target: x86_64-pc-linux-gnu
[forge@nuc7i7bnh sg-uc]$ marten -v
Marten 0.1.2
```

**Comment by ellmetha** [comment]
Hey! 👋 

Thanks for reporting this!

Regarding `--error-trace`, this option is simply supposed to output the whole compilation error trace for compilation errors (as a matter of fact, this is simply a wrapper around the standard `--error-trace` compiler option that you can use when running commands like `crystal build` for example).

The error you mentioned seems to occur because you're trying to access the app using `http://0.0.0.0:8000` (instead you should use `http://127.0.0.1:8000` in development environments). That being said something is clearly off because the CLI was outputting `0.0.0.0:8000`, so the development host that is used for newly generated projects is probably incorrect. But in theory, the framework should've rendered a "bad request" error page (because the URL you are using to access the app is not part of the [allowed hosts](https://martenframework.com/docs/development/reference/settings#allowed_hosts)). 

After further investigation, there were a few problems explaining the issue you were experiencing:

1. first, the host used for development in projects generated with the `new` management command should be something like `127.0.0.1` or `localhost` (as mentioned previously)
2. since this was not the case, the framework was trying to render a "bad request" page but this was failing because the bad request handler was trying to find a template to render, and by doing so it was trying to look for the request's flash store (but this one wasn't initialized because of (1) and the fact that the associated middleware was not yet executed)

(1) was fixed in 42f9628d594729cd3051856e2dd5193987ed0459 and (2) was fixed in 9a3fe657010449d9a9b8ebdd90381858dbba7f7a. These two fixes will part of the upcoming `0.1.3` bugfix release.

Thanks again for reporting this and helping improve the framework! 🙏

**Comment by ellmetha** [comment]
Oh, I forgot about this:

> I also noticed that marten serve --error-trace does not "hot reload" or "watch" changed files.
>
> Adding my [first handler](https://martenframework.com/docs/getting-started/tutorial#writing-a-first-handler) did not change anything.

I quickly tested and I wasn't able to reproduce this with my local setup. If the problem remains, feel free to create another issue specifically related to that! 🙂
